src/mqtt_client/mod.rs

```diff
@@ -199,11 +229,56 @@ where
             mqtt,
             state: sm::StateMachine::new(sm::Context::new(clock)),
             settings,
-            settings_prefix,
             prefix: String::from(prefix),
+            listing_state: None,
         })
     }
 
+    fn handle_listing(&mut self) {
+        let Some((ref props, ref mut iter)) = &mut self.listing_state else {
+            return;
+        };
+
+        let props = minimq::types::Properties::DataBlock(props);
+
+        let mut data = [0u8; MESSAGE_SIZE];
+
+        while self.mqtt.client().can_publish(QoS::AtLeastOnce) {
+            // Note(unwrap): Publishing should not fail because `can_publish()` was checked before
+            // attempting this publish.
+            let response: Response<MAX_TOPIC_LENGTH> = iter
+                .next()
+                .map(|path| Response::custom(ResponseCode::Continue, &path))
+                .unwrap_or_else(Response::ok);
+
+            let message = match serde_json_core::to_slice(&response, &mut data) {
+                Ok(len) => &data[..len],
+
+                // If the message buffer is too small, we can't ever provide a response.
+                _ => return,
```

Comment:
Not error here?

Reply:
The `poll()` closure currently returns `()`, so there's no chance to return an error. We could refactor it, but I'd rather keep that into a separate PR, since there's other rework around the `poll()` function we could complete as well. I think this is maintaining current behavior.

Reply:
I meant still providing a fixed error response to the user. But I'm unsure whether it's better to abort a listing when we hit an error or silently discard paths that are too long.

Reply:
If we can't enqueue a response of 32-bytes, it doesn't make sense to e.g. try something progressively smaller until it eventually works. I would just say that we have a static limit on the minimum transport size

Reply:
Ah hmm, this is for the list command, I was thinking this was in generic `get/set/list` responses.

Indeed, we likely need to generate an error if we could not serialize the response into the buffer, since CD or response topic may be taking too much space.